### PR TITLE
algorithm: unique paths 2

### PR DESCRIPTION
**Pull request by Saimon398**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1257"></a>

### CODE REVIEW

File: Dynamic-Programming/UniquePaths2.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * @description Return 'rows x columns' grid with cells filled by 'filler'
+ * @param {Number} rows Number of rows in the grid
+ * @param {Number} columns Number of columns in the grid
+ * @param {String | Number | Boolean} filler The value to fill cells
+ * @returns {Array [][]}
+ */
+const generateMatrix = (rows, columns, filler = 0) => {
+  const matrix = []
+  for (let i = 0; i < rows; i++) {
+    const submatrix = []
```
Comment (appgurueu) [inline Dynamic-Programming/UniquePaths2.js:27]: row

File: Dynamic-Programming/UniquePaths2.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    for (let k = 0; k < columns; k++) {
+      submatrix[k] = filler
```
Comment (appgurueu) [inline Dynamic-Programming/UniquePaths2.js:29]: Might as well use `Array.fill` here

**Review by appgurueu — APPROVED** [review]
Looks fine

**Comment by raklaptudirm** [comment]
@appgurueu Should I wait for the conversations to be resolved?

**Comment by Saimon398** [comment]
Done

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**